File: diagonalnegativos.py



n = int(input("Qual a ordem da matriz? "))

mat = [[0 for x in range (n)] for x in range (n)]

for i in range (0, n):
    for j in range (0, n):
        mat[i][j] = int(input(f"Elemento [{i}, {j}]: "))

print()
print("Diagonal principal: ")

for i in range (0, n):
    for j in range (0, n):
        if i == j:
            print(mat[i][j], " ", end = "")


soma = 0

for i in range (0, n):
    for j in range (0, n):
        if mat[i][j] < 0:
            soma += 1

print()
print(f"Quantidade de negativos = {soma}")            

File: Projetos em C++/diagonal_negativos.cpp
# include <bits/stdc++.h>

using namespace std;

int main() {

    int n, soma;

    cout << "Qual a ordem da matriz? ";
    cin >> n;

    int mat[n][n];

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            cout << "Elemento [" << i << ", " << j << "]: ";
            cin >> mat[i][j];
        }
    }
    
    cout << endl << "Diagonal principal: " << endl;

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (i == j) {
                cout << mat[i][j] << " ";
            }
        }
    }

    soma = 0;

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (mat[i][j] < 0) {
                soma += 1;
            }
        }
    }

    cout << endl << "Quantidade de negativos = " << soma; 

    return 0;
}